services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/mapper/StaticImmutableTableSchema.java

```diff
@@ -208,6 +210,7 @@ private StaticImmutableTableSchema(Builder<T, B> builder) {
         this.buildItemFunction = builder.buildItemFunction;
         this.tableMetadata = tableMetadataBuilder.build();
         this.itemType = EnhancedType.of(builder.itemClass);
+        this.preserveEmtpyBean = builder.itemClass.getAnnotation(DynamoDbPreserveEmptyBean.class) != null;
```

Comment:
Static*TableSchema must not perform any reflection. It must be statically configurable from its builder using properties and tags. We could achieve this by having a boolean builder property for example like 
```
   .preserveEmptyBean(true)
```

and then BeanTableSchema and ImmutableTableSchema would scan for the annotation and add that flag to the StaticImmutableTableSchema they were building.

Reply:
Yeah, I was not aware of this. I'll see if I can add a checkstyle for this.

Reply:
It's more of a design thing than a technical limitation. StaticTableSchema and StaticImmutableTableSchema are both public classes (APIs) that customers that specifically do not want to use reflection can opt to use to construct their TableSchema instances without any runtime class metadata being involved. We wouldn't want to break that contract/promise.